x/capability/keeper/keeper_test.go

```diff
@@ -240,6 +240,37 @@ func (suite *KeeperTestSuite) TestReleaseCapability() {
 	suite.Require().Nil(got)
 }
 
+func (suite KeeperTestSuite) TestRevertCapability() {
```

Comment:
Thought about how to test this. Could have Delivered a tx to simapp that fails after claiming a capability and test that way. However, that would require importing a bunch of ibc packages and a lot of setup to test so it wasn't worth it.

Decided to just mock what `runMsgs` would do here, and show that capability will not be retrievable unless the cached context gets written after successful tx execution

Reply:
As long as this works and tests the cases we need. I initially imagined we'd add a testcase to IBC or some other module that actually has a message handler that uses the x/capability module.